src/Sylius/Behat/Page/Admin/DashboardPage.php

```diff
@@ -13,91 +13,141 @@
 {
+    /**
+     * @template TKey of array-key
+     * @template TValue
+     *
+     * @param array<TKey, TValue>|\ArrayAccess<TKey, TValue> $minkParameters
+     */
     public function __construct(
         Session $session,
-        $minkParameters,
+        array|\ArrayAccess $minkParameters,
         RouterInterface $router,
         protected TableAccessorInterface $tableAccessor,
     ) {
         parent::__construct($session, $minkParameters, $router);
     }
 
+    /** @throws ElementNotFoundException */
     public function getTotalSales(): string
     {
         return $this->getElement('total_sales')->getText();
     }
 
+    /** @throws ElementNotFoundException */
     public function getNumberOfNewOrders(): int
     {
         return (int) $this->getElement('new_orders')->getText();
     }
 
+    /** @throws ElementNotFoundException */
     public function getNumberOfNewOrdersInTheList(): int
     {
         return $this->tableAccessor->countTableBodyRows($this->getElement('order_list'));
     }
 
+    /** @throws ElementNotFoundException */
     public function getNumberOfNewCustomers(): int
     {
         return (int) $this->getElement('new_customers')->getText();
     }
 
+    /** @throws ElementNotFoundException */
     public function getNumberOfNewCustomersInTheList(): int
     {
         return $this->tableAccessor->countTableBodyRows($this->getElement('customer_list'));
     }
 
+    /** @throws ElementNotFoundException */
     public function getAverageOrderValue(): string
     {
         return $this->getElement('average_order_value')->getText();
     }
 
+    /** @throws ElementNotFoundException */
     public function getSubHeader(): string
     {
         return trim($this->getElement('sub_header')->getText());
     }
 
+    /** @throws ElementNotFoundException */
     public function isSectionWithLabelVisible(string $name): bool
     {
         return $this->getElement('admin_menu')->find('css', sprintf('div:contains(%s)', $name)) !== null;
     }
 
+    /** @throws ElementNotFoundException */
     public function logOut(): void
     {
         $this->getElement('logout')->click();
     }
 
+    /** @throws ElementNotFoundException */
     public function chooseChannel(string $channelName): void
     {
         $this->getElement('channel_choosing_link', ['%channelName%' => $channelName])->click();
     }
 
+    /** @throws ElementNotFoundException */
+    public function chooseYearSplitByMonthsInterval(): void
+    {
+        $this->getElement('year_split_by_months_statistics_button')->click();
+    }
+
+    /** @throws ElementNotFoundException */
+    public function chooseMonthSplitByDaysInterval(): void
+    {
+        $this->getElement('month_split_by_days_statistics_button')->click();
+    }
+
+    /** @throws ElementNotFoundException */
+    public function choosePreviousPeriod(): void
+    {
+        $this->getElement('navigation_previous')->click();
+
+        usleep(500000);
```

Comment:
Why 500000? Could we possibly adopt an approach similar to what is used in `JavaScriptTestHelper`

Reply:
I'm not certain, as the element is consistently present when selenium (I hope it's selenium) clicks it, and it retrieves the value properly, but not the correct one, so I opted to insert this slight delay

Reply:
It isn't Selenium. We're using chromedriver.

Reply:
Anyway, I have mixed feeling about it, but no better idea for now.

Reply:
You could try using `JQueryHelper::waitForAsynchronousActionsToFinish` or wait until the `active` class disappears from the `stats-loader` element; and as a side note, why usleep? It looks so weird having half a million in code